src/app/basemap/basemap.component.ts

```diff
@@ -39,5 +43,16 @@ export class BasemapComponent implements OnInit {
       pitch: 60,
       center: this.center
     });
+
+    const layers: [object] = this.moduleHandler.getLayers();
```

Comment:
This returns:
<img width="679" alt="Screen Shot 2019-03-28 at 18 59 01" src="https://user-images.githubusercontent.com/14296774/55198239-a7db9380-518b-11e9-9abe-db4fb5fbce62.png">

I guess the way I saw this injection is that layer could be dynamically added to a 'global' map instance, preferably after init stage

Reply:
the addLayer function is wrapped inside the `this.map.on(‘load’, callback)` on 284e4f9fa21d8e84b02ce16148168126c03032c7. I’m missing the subscription of the data, may relate to #6?

> the layer could be dynamically added to a ‘global' map instance, preferably after init stage

does this mean, by looking at `basemap.component.ts` we can no longer see whats added or not? I don’t know if mutating basemapComponent’s property from a different instance will give clarity. The map does not need to know what is added, but we might want to when it is added.